app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt

```diff
@@ -0,0 +1,32 @@
+package com.wire.android.di
+
+import android.content.Context
+import com.wire.android.util.DeviceLabel
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.feature.auth.LoginUseCase
+import dagger.hilt.android.components.ViewModelComponent
+import dagger.hilt.android.scopes.ViewModelScoped
+import javax.inject.Singleton
+
+
+@Module
+@InstallIn(ViewModelComponent::class)
+class CoreLogicModule {
+
+    @ViewModelScoped
+    @Provides
+    fun coreLogicProvider(@ApplicationContext context: Context): CoreLogic {
```

Comment:
I'd say CoreLogic can - and maybe should - be a singleton. Because inside of it, we might want to have singletons in its injection mecanism.